### PR TITLE
Add power-saving and max-performance settings

### LINKED ISSUE #91: Does anyone need a Low Power Profile for Cable or a High Power Profile for Battery?
Hi folks,

I just found out what these AC/DC Power source commands are doing.
There are two of them, one for Power over Cable `0x11` and one for Power over Battery `0x12`

These get automatically applied on plug or unplug charging cable.

I figured out what these things do change. Both can be applied on cable and both can be applied on battery. So you can switch to high performance on battery or switch to maximum power savings while on cable.

0x11 Cable Mode
- change all CLK Setpoint Values to 50
- Infinity Fabric is nearly pinned, even in Idle with power saving profile on my Rainor It down clocks only to a minimum of MAXCLK-80Mhz (1120Mhz)
 
0x12 Battery Mode
- change all CLK Setpoint Values to 95
- Minimum Infinity Farbic speed of 400MHz at idle but still boosting to max

 CLK Setpoint does affect how the frequency response to cpu load.

Things we all know is the importance of the Infinity Fabric, but what I didn't know is the high power draw.
At Idle this downclock to 400Mhz does save 50% Package Power. It goes from 4W to 2W.

I know that there is a Lenovo Platform Profile for power management. And I guess I am using it correctly, because Lenovo software is telling me that the Fan profile will respond to windows power plan changes. If this is really true, then I can tell you that this battery mode command does save more power than the normal windows power saving profile and Lenovo platform profile.

### But here is the question for you, would anybody request something like:
- I want to change the word, give me these 2W of energy saving while on Cable
- Even on battery I want maximum snappiness, High Performance Power Plan is not enough! Don't care if the battery lasts only 3 instead of 3.5 hours

If yes, then we could add these weird commands to ryzenAdj. But I don't want to add unnecessary commands to avoid even more confusion.

We could call it --save-the-world as boolean in a special release ;)

### PR DESCRIPTION
**Pull request by Falcosc**
to control hidden options like boost delay on battery
Resolves #91

### CODE REVIEW

**Comment by FlyGoat** [comment]
Hmmm, as it's just AC/DC mode SMU commands, what about call it `--force-dc-mode` or sth?

**Comment by Falcosc** [comment]
I thought about it, but this could be confusing because it does not change how the power source gets reported to the OS.

This is the lost of information which I would like to press into this CLI description:
https://www.reddit.com/r/Amd/comments/mgmxjk/ryzenadj_we_did_discover_a_way_to_remove_boost/gswmzju/?utm_source=reddit&utm_medium=web2x&context=3

File: README.md [hunk 1]
```diff
@@ -55,6 +55,8 @@ Settings
     --dgpu-skin-temp=<u32>                dGPU Skin Temperature Limit   - STT LIMIT dGPU (degree C)
     --apu-slow-limit=<u32>                APU PPT Slow Power limit for A+A dGPU platform - PPT LIMIT APU (mW)
     --skin-temp-limit=<u32>               Skin Temperature Power Limit (mW)
+    --power-saving                        Hidden options to improve power efficiency (is set when AC unplugged): behavior depends on CPU generation, Device and Manufacture
+    --max-performance                     Hidden options to improve performance (is set when AC plugged in): behaivor depends on CPU generation, Device and Manufacture
```
Comment (Falcosc) [inline README.md:59]: typo

**Comment by Falcosc** [comment]
```
--force-dc-mode        Sets hidden options to maybe save power: behavior depends on CPU generation, Device and Vendor
--force-ac-mode        Sets hidden options to maybe improve performance: behavior depends on CPU generation, Device and Vendor
```

**Comment by FlyGoat** [comment]
Oh I'll double check what it actually does.

**Comment by Falcosc** [comment]
It does set 4 of these setpoint values to 95 for battery
And 50 for AC

**Comment by Falcosc** [comment]
Raven Ridge does support that too. I did successfully enable the boost delay on a 2400G

**Comment by Falcosc** [comment]
@FlyGoat people hate our cli option name.

Wish number 1: --boost-delay=[true|false]
Wish number 2: anything with performance or power saving in the name

I did argue that both wishes don't fit an AC/DC Power profile enablement if you don't know what's inside this power profile.
But they hate anything which does include AC/DC. It was suggested to call them battery and wall.
I don't like wall at all. So at the end my compromise was going for performance/power saving.

**Comment by FlyGoat** [comment]
ok, then LGTM, thanks!

**Comment by FlyGoat** [comment]
Probably it's time to make a release?

**Comment by Falcosc** [comment]
I will fix some typos, add some more tricks to the PowerShell script documentation, update the tuning guide, update the faq, then I will make a release.

But we could create a prerelease if you want to share it now :)

If not, I will release it beginning of next week together with a nice benchmark chart for 6 different systems (2x ryzen 4000, 2x ryzen 2000, 2x devices with no boost at all)

I'm still looking for at least 2 scores of ryzen 3000 for my release celebration chart. Should get them this evening.

**Pull request merged**